### ISSUE TITLE
Support for go dep with purego (purego branch instead of tag?)

### ISSUE THREAD

**Issue by wirepair** [cznic/zappy#7]
I use go dep (https://golang.github.io/dep/) to manage my packages, and wish to support windows with ql. Unfortunately, zappy's master branch requires gcc. While you can build with -tags purego this doesn't seem to work with the dep command. I believe if the purego version was it's own branch i could do 
```
dep ensure -add github.com/cznic/zappy@^purego
```
And things would most likely work. Not sure if you plan on supporting go dep but if you do this ticket is to capture the requirements.
Thank you!

**Comment by cznic** [comment]
@wirepair 

Please test the branch approach locally and if it works for you please send a PR including a new changelog section in the `README.md` files of both branches and your name and mail added to the `AUTHORS` and/or `CONTRIBUTORS` files also in both branches, thank you.

**Comment by wirepair** [comment]
Hello @cznic I did a bit more research into this before I tried anything. So it seems like dep does not support build tags yet (https://github.com/golang/dep/issues/291). However, I realized I can just pass "-tags purego" to *my* project and it will build properly. Even if we split this out to its own branch, we'd probably have to delete the cgo version, which would be annoying having the branches differ. I think it makes more sense to leave it as-is for now, and wait to see how dep plans on supporting build tags. 

To re-iterate to anyone else viewing this ticket: 
You'll need to build your project with ```go build -tags purego``` if you import zappy or ql when you use dep ensure.
